### DISCUSSION TITLE
Supports directory mounts for model types provided by extensions

### DISCUSSION THREAD

**Discussion by KagurazakaNyaa** [AbdBarho/stable-diffusion-webui-docker#380] [Ideas]
Some extensions will create other model directories under the model directory, such as Deforum(#255) and dreambooth(https://github.com/d8ahazard/sd_dreambooth_extension/blob/main/dreambooth/shared.py#L306)
In order to support the persistence of these models, we have to manually mount them in `data/config/auto/startup.sh`, and it is easy to miss the model types of newly installed extensions in the future.
So why don't we just persist the entire `models` directory to `/data/models` to solve this problem?

like this
https://github.com/KagurazakaNyaa/stable-diffusion-webui-docker/commit/5d9f3daa1ff48685ac8b6e6823f8921d5607f482

**Comment by AbdBarho** [comment]
This works elegantly for `auto`, but maybe not as elegant for the others.

**Comment by KagurazakaNyaa** [comment]
Maybe we can make it compatible with other types of webui by creating a directory alias under `data/models`.
But I doubt that anyone will use multiple different webuis in the same directory at the same time.
Maybe my attempt in this commit can be referred to to solve this problem.

https://github.com/KagurazakaNyaa/stable-diffusion-webui-docker/commit/d1e5d5905902b4413efc051edae2d8388d573170
